pkg/subctl/cmd/gather/logs.go

```diff
@@ -62,41 +61,55 @@ func gatherPodLogsByContainer(podLabelSelector, container string, info Info) {
 	}
 }
 
-func podLogsToFile(pod *corev1.Pod, podLogOptions *corev1.PodLogOptions, info Info) error {
-	logRequest := info.ClientSet.CoreV1().Pods(pod.Namespace).GetLogs(pod.Name, podLogOptions)
+func outputPodLogs(pod *corev1.Pod, podLogOptions corev1.PodLogOptions, info Info) error {
+	var outputPodLogsErrors []string
 
-	logs, err := processLogStream(logRequest)
+	outputPreviousPodLogErr := outputPreviousPodLog(pod, podLogOptions, info)
+	if outputPreviousPodLogErr != nil {
+		outputPodLogsErrors = append(outputPodLogsErrors, fmt.Sprintf("error %s getting logs from previous pod %s",
+			outputPreviousPodLogErr.Error(), pod.Name))
+	}
+
+	outputCurrentPodLogErr := outputCurrentPodLog(pod, podLogOptions, info)
+	if outputCurrentPodLogErr != nil {
+		outputPodLogsErrors = append(outputPodLogsErrors, fmt.Sprintf("error %s getting logs from current pod %s",
+			outputCurrentPodLogErr.Error(), pod.Name))
+	}
+
+	var outputPodLogsError string
+	for _, err := range outputPodLogsErrors {
+		outputPodLogsError += err + "\n" // Prints each error in a new line
+	}
+	return errors.New(outputPodLogsError)
```

Comment:
Rather than concatenating errors, I suggest to simplify it and just output the errors directly:

```
func outputPodLogs(pod *corev1.Pod, podLogOptions corev1.PodLogOptions, info Info) {
    err := outputPreviousPodLog(pod, podLogOptions, info)
    if err != nil {
        info.Status.QueueFailureMessage(fmt.Sprintf("Error outputting previous log for pod %q: %v", pod.Name, err))		
    }

    err := outputCurrentPodLog(pod, podLogOptions, info)
    if err != nil {
        info.Status.QueueFailureMessage(fmt.Sprintf("Error outputting current log for pod %q: %v", pod.Name, err))
    }
}
```